CHANGES/7694.feature

```diff
@@ -0,0 +1 @@
+Sped up repository pre-migration by skipping the relations pre-migration when nothing changed in a repository.
```

Comment:
s/ relations/ repository content relations

Reply:
i would probably use a .bugfix here or .misc, it does not bring new functionality

Reply:
Interesting :), I think about it from the other side. It's not a bug fix, it's a perf improvement :) and users should be aware of it, so I suggest not to use .misc.
I can convert it to a bug fix.

Reply:
for the visibility reasons I prefer to convert it to the bugfix. I don't think any our previous perf improvements went as .feature :)

Reply:
Actually, most of the previous performance improvements didn't come with changelog entries at all :)  Except when it was caused by an actual bug with user-facing impacts outside of performance.

I'm fine with all of the above including .misc.  Do you have a ballpark % of how much it improved things in your testing?

Reply:
@dralley , we should advertise perf improvements as much as we can. This is what I heard from @melcorr :) So let's try to make it visible in the changelog, wdyt?

Performance improvement here is for a specific case when a migration has been interrupted. Which we expect users to do if they migrate a large system. Those repositories which have been pre-migrated but not migrated yet, will no longer be pre-migrated one more time. In such cases pre-migration part went from 20-40 mins to 1-2 min, depending on the hardware.

If all repos has been migrated and nothing changed, this patch doesn't improve performance because we check if repo.is_migrated before pre-migrating it again.

Reply:
> In such cases pre-migration part went from 20-40 mins to 1-2 min, depending on the hardware.

Cool, in that case it's definitely worth advertising.